sqlconnect/internal/redshift/driver/rows.go

```diff
@@ -0,0 +1,120 @@
+package driver
+
+import (
+	"context"
+	"database/sql/driver"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/service/redshiftdata"
+	"github.com/aws/aws-sdk-go-v2/service/redshiftdata/types"
+)
+
+type redshiftRows struct {
+	ctx context.Context
+	id  string
+	p   *redshiftdata.GetStatementResultPaginator
+
+	page        *redshiftdata.GetStatementResultOutput
+	columns     []types.ColumnMetadata
+	columnNames []string
+	index       int
+}
+
+func newRows(ctx context.Context, id string, p *redshiftdata.GetStatementResultPaginator) (*redshiftRows, error) {
+	debugLogger.Printf("[%s] create rows", id)
+	rows := &redshiftRows{
+		ctx: ctx,
+		id:  id,
+		p:   p,
+	}
+	return rows, rows.getStatementResult()
+}
+
+func (rows *redshiftRows) Close() (err error) {
+	debugLogger.Printf("[%s] rows close called", rows.id)
+	return nil
+}
+
+func (rows *redshiftRows) Columns() []string {
+	return rows.columnNames
+}
+
+func (rows *redshiftRows) ColumnTypeDatabaseTypeName(index int) string {
+	return strings.ToUpper(*rows.columns[index].TypeName)
+}
+
+func (rows *redshiftRows) getStatementResult() error {
+	if rows.p == nil {
+		return io.EOF
+	}
+	var err error
+	rows.page, err = rows.p.NextPage(rows.ctx)
+	if err != nil {
+		return err
+	}
+	rows.columns = rows.page.ColumnMetadata
+	rows.columnNames = make([]string, 0, len(rows.columns))
+	for _, meta := range rows.columns {
+		rows.columnNames = append(rows.columnNames, *meta.Name)
+	}
+	rows.index = 0
+	return nil
+}
+
+func (rows *redshiftRows) Next(dest []driver.Value) error {
+	debugLogger.Printf("[%s] rows next called", rows.id)
+	if rows.page == nil || rows.index >= len(rows.page.Records) {
+		if !rows.p.HasMorePages() {
+			return io.EOF
+		}
+		if err := rows.getStatementResult(); err != nil {
+			return err
+		}
+		rows.index = 0
+		if len(rows.page.Records) == 0 {
+			return io.EOF
+		}
+	}
+	record := rows.page.Records[rows.index]
+	for i := range dest {
+		if i < len(record) {
+			switch field := record[i].(type) {
+			case *types.FieldMemberIsNull:
+				dest[i] = nil
+			case *types.FieldMemberStringValue:
+				switch {
+				case strings.EqualFold(*rows.page.ColumnMetadata[i].TypeName, "timestamp"):
```

Comment:
**Note:** to be compatible with the behaviour of pg driver, we need to:
- Parse the string into a time
- Convert the time to UTC and parse it as an RFC3339 string.